### DISCUSSION TITLE
Group search results by showing the document with the max or min value of a field

### DISCUSSION THREAD

**Discussion by malsioufi** [appbaseio/reactivesearch#1636] [Q&A]
I have checked this tutorial on the documentation of ReactiveSearch.
https://docs.appbase.io/docs/reactivesearch/v3/advanced/groupingresults/

I understand the idea of having `aggregationField `prop of `ReactiveList `component to get aggregated data, but is there anyway to decide which document to return?
For example, in the same mentioned example in the documentation, is it possible to show distinct value using the field "brand" but also to show the document that has the max value or min value of for example field "price" instead of showing a random one. 

In Elasticsearch that is possible by this query:
```
{
  "aggs": {
    "2": {
      "terms": {
        "field": "brand.keyword",
        "order": {
          "_count": "desc"
        },
        "size": 50
      },
      "aggs": {
        "1": {
          "max": {
            "field": "price"
          }
        }
      }
    }
  },
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "match_all": {}
        }
      ]
    }
  }
}
```
Is that doable in ReactiveSearch?

**Comment by siddharthlatest** [comment]
@malsioufi This is now possible using `distinctField`, it requires the use of appbase.io service as it relies on server-side query generation.

![](https://i.imgur.com/15PSjwA.png)

Read more in the docs over at: https://docs.appbase.io/docs/reactivesearch/v3/result/reactivelist/
